auth/service.go

```diff
@@ -956,3 +961,122 @@ func DecodeDomainUserID(domainUserID string) (string, string) {
 		return "", ""
 	}
 }
+
+func (svc service) DeleteEntityPolicies(ctx context.Context, entityType, id string) (err error) {
+	switch entityType {
+	case ThingType:
+		// Remove policy of groups
+		req := PolicyReq{
+			SubjectType: GroupType,
+			Object:      id,
+			ObjectType:  ThingType,
+		}
+
+		if err := svc.DeletePolicy(ctx, req); err != nil {
+			return err
+		}
+
+		// Remove policy from domain
+		req.SubjectType = DomainType
+		if err := svc.DeletePolicy(ctx, req); err != nil {
+			return err
+		}
+
+		// Remove policy of users
+		req.SubjectType = UserType
+		if err := svc.DeletePolicy(ctx, req); err != nil {
+			return err
+		}
+
+		return nil
+	case UserType:
+		domainsPage, err := svc.domains.ListDomains(ctx, Page{SubjectID: id, Limit: defLimit})
+		if err != nil {
+			return err
+		}
+
+		if domainsPage.Total > defLimit {
+			for i := defLimit; i < int(domainsPage.Total); i += defLimit {
+				page := Page{SubjectID: id, Offset: uint64(i), Limit: defLimit}
+				dp, err := svc.domains.ListDomains(ctx, page)
+				if err != nil {
+					return err
+				}
+				domainsPage.Domains = append(domainsPage.Domains, dp.Domains...)
+			}
+		}
+
+		for _, domain := range domainsPage.Domains {
+			policy := PolicyReq{
+				Subject:     EncodeDomainUserID(domain.ID, id),
+				SubjectType: UserType,
+				Object:      domain.ID,
+				ObjectType:  DomainType,
+			}
+			if err := svc.agent.DeletePolicy(ctx, policy); err != nil {
+				return err
+			}
+		}
+
+		req := PolicyReq{
+			Subject:     id,
+			SubjectType: UserType,
+			ObjectType:  ThingType,
+		}
+		if err := svc.agent.DeletePolicy(ctx, req); err != nil {
+			return err
+		}
+		req.ObjectType = GroupType
+		if err := svc.agent.DeletePolicy(ctx, req); err != nil {
+			return err
+		}
+		req.ObjectType = DomainType
+		if err := svc.agent.DeletePolicy(ctx, req); err != nil {
+			return err
+		}
```

Comment:
the below part is no needed. 
There will be no subject just with user id 
User's Subject id in spice will be always combination <domain_id>_<user_id>

```go
		req := PolicyReq{
			Subject:     id,
			SubjectType: UserType,
			ObjectType:  ThingType,
		}
		if err := svc.agent.DeletePolicy(ctx, req); err != nil {
			return err
		}
		req.ObjectType = GroupType
		if err := svc.agent.DeletePolicy(ctx, req); err != nil {
			return err
		}
		req.ObjectType = DomainType
		if err := svc.agent.DeletePolicy(ctx, req); err != nil {
			return err
		}
```

Reply:
If we leave it spiceDB will see it as a valid policy and will not clean up